samples/sampler_object.html

```diff
@@ -0,0 +1,231 @@
+<!DOCTYPE html>
+<!-- Ported from the OpenGL Samples Pack https://github.com/g-truc/ogl-samples/blob/master/tests/gl-330-sampler-object.cpp -->
+<html lang="en">
+
+<head>
+    <title>WebGL 2 Samples - sampler_object</title>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0">
+    <link rel="stylesheet" href="style.css">
+    <script src="utility.js"></script>
+</head>
+
+<body>
+    <div id="info">WebGL 2 Samples - sampler_object</div>
+
+    <script id="vs" type="x-shader/x-vertex">
+        #version 300 es
+        #define POSITION_LOCATION 0
+        #define TEXCOORD_LOCATION 4
+
+        precision highp float;
+        precision highp int;
+
+        uniform mat4 mvp;
+
+        layout(location = POSITION_LOCATION) in vec2 position;
+        layout(location = TEXCOORD_LOCATION) in vec2 textureCoordinates;
+
+        out vec2 st;
+
+        void main()
+        {
+            st = textureCoordinates;
+            gl_Position = mvp * vec4(position, 0.0, 1.0) ;
+        }
+    </script>
+
+    <script id="fs" type="x-shader/x-fragment">
+        #version 300 es
+        #define FRAG_COLOR_LOCATION 0
+
+        precision highp float;
+        precision highp int;
+        layout(std140, column_major) uniform;
+
+        struct Material
+        {
+            sampler2D diffuse[2];
+        };
+
+        uniform Material material;
+
+        in vec2 st;
+
+        layout(location = FRAG_COLOR_LOCATION) out vec4 color;
+
+        void main()
+        {
+            if (st.y / st.x < 1.0) {
+                color = texture(material.diffuse[0], st);
+            } else {
+                color = texture(material.diffuse[1], st) * .77;
```

Comment:
I suggest using `0.77` (just like there is `0.0`, not `.0`, in the vertex shader), but some folks do just `.77`.  I don't think it is as popular, but it is valid GLSL.